lib/md055.js

```diff
@@ -0,0 +1,40 @@
+// @ts-check
+
+"use strict";
+
+const { addError} = require("../helpers");
+
+const noLeadingPipeErrorMessage = "Table rows must start with a pipe";
+const noTrailingPipeErrorMessage = "Table rows must end with a pipe";
+
+module.exports = {
+    "names": ["MD055", "table-missing-border"],
+    "description": "Table is missing border pipes",
+    "information": new URL("https://github.github.com/gfm/#tables-extension-"),
+    "tags": ["tables"],
+    "function": function MD055(params, onError) {
+        const outerPipes = String(params.config.style || "both");
+        const needLeading = outerPipes === "both" || outerPipes === "leading";
+        const needTrailing = outerPipes === "both" || outerPipes === "trailing";
+
+        const tables = params.parsers.markdownit.tokens.filter(
+            (token) => token.type === "table_open"
+        );
+
+        for (const table of tables) {
+            const [tableStart, tableEnd] = table.map;
+            const lines = params.lines;
+            const indentation = params.lines[tableStart].indexOf("|");
```

Comment:
But good news with micromark is that you shouldn't have to do unreliable line parsing like this as there tend to be tokens for everything.